syne_tune/experiments/results_utils.py

```diff
@@ -10,14 +10,364 @@
+    that side. This feature is useful to filter out results of erroneous
+    attempts.
+
+    If ``seed_key`` is given, the returned index is a dictionary with keys
+    ``(benchmark_name, seed)``, where ``seed`` is the value corresponding to
+    ``seed_key`` in the metadata dict. This mode is needed for plots focusing
+    on a single experiment.
+
+    :param experiment_names: Tuple of experiment names (prefixes, without the
+        timestamps)
+    :param metadata_to_setup: See above
+    :param metadata_to_subplot: See above. Optional
+    :param metadata_keys: See above. Optional
+    :param metadata_subplot_level: See above. Defaults to ``False``
+    :param benchmark_key: Key for benchmark in metadata files. Defaults to
+        "benchmark"
+    :param with_subdirs: See above. Defaults to "*"
+    :param datetime_bounds: See above
+    :param seed_key: See above
+    :return: Dictionary; entry "index" for index (see above); entry
+        "setup_names" for setup names encountered; entry "metadata_values" see
+        ``metadata_keys``
+    """
+    reverse_index = dict()
+    setup_names = set()
+    metadata_values = dict()
+    if metadata_keys is None:
+        metadata_keys = []
+    if metadata_to_subplot is None:
+        metadata_subplot_level = False
+    datetime_bounds = _convert_datetime_bounds(datetime_bounds, experiment_names)
+    is_map_dict = isinstance(metadata_to_setup, dict)
+    for experiment_name in experiment_names:
+        datetime_lower, datetime_upper = datetime_bounds[experiment_name]
+        patterns = [experiment_name + "-*/" + ST_METADATA_FILENAME]
+        if with_subdirs is not None:
+            if not isinstance(with_subdirs, list):
+                with_subdirs = [with_subdirs]
+            pattern = patterns[0]
+            patterns = [experiment_name + f"/{x}/" + pattern for x in with_subdirs]
+        logger.info(f"Patterns for result files: {patterns}")
+        for meta_path in itertools.chain(
+            *[experiment_path().glob(pattern) for pattern in patterns]
+        ):
+            tuner_path = meta_path.parent
+            if _skip_based_on_datetime(tuner_path.name, datetime_lower, datetime_upper):
+                continue  # Skip this result
+            # Load metadata
+            try:
+                with open(str(meta_path), "r") as f:
+                    metadata = json.load(f)
+            except FileNotFoundError:
+                metadata = None
+            if metadata is None:
+                continue
+            benchmark_name = _get_benchmark_name(benchmark_key, metadata, tuner_path)
+            seed_value = None if seed_key is None else int(metadata[seed_key])
+            try:
+                # Extract ``setup_name``
+                map = (
+                    metadata_to_setup[benchmark_name]
+                    if is_map_dict
+                    else metadata_to_setup
+                )
+                setup_name = map(metadata)
```

Comment:
I don't understand what setup_name here exactly is? Is that just a identifier of a single run for one method?

Reply:
A setup is a bit abstract, it translates to essentially a curve of a certain color. The curves in the same plots are for different setups, but the same setup can appear in different subplots.

Often, setup equates to "HPO method", but sometimes it does not. You can also use plotting, running the same method on different variations (e.g., n_workers; config space), and then a setup is not a method.

One example would be the plots you did, where you compare ASHA with 2, 4, 8, 16 workers. The setup would be n_workers then.